### DISCUSSION TITLE
Is there a way to retrieve key-value cache values using onnxruntime-genai?

### DISCUSSION THREAD

**Discussion by Tabrizian** [microsoft/onnxruntime-genai#416] [New features / APIs]
Can we retrieve the key-value cache values using this package for re-use across different generations?

**Comment by natke** [comment]
Hi @Tabrizian, this is an API that we are looking at for another customer. I will move it into the Discussions forum where we can continue the conversation. If you have specific use cases, please feel free to add them there!

**Comment by Tabrizian** [comment]
Thanks @natke. The use-case I have is to be able to manually control which set of requests are included in every batch. Being able to obtain key-value cache parameters can help us avoid redundant computation when swapping sequences in and out of the request batch.
